launcher/minecraft/PackProfile.cpp

```diff
@@ -954,8 +988,58 @@ bool PackProfile::setComponentVersion(const QString& uid, const QString& version
         // set existing
         if (component->revert()) {
+            // remove linked components to let them re-resolve their versions
+            if (important) {
+                component->waitLoadMeta();
+                auto linked = collectTreeLinked(uid);
+                for (auto comp : linked) {
+                    if (comp->isCustom()) {
+                        continue;
+                    }
+                    if (modloaderMapping.contains(comp->getID())) {
+                        qCDebug(instanceProfileC)
+                            << d->m_instance->name() << "|" << comp->getID() << "is a mod loader, updating it to a compatible version";
+
+                        auto versionList = APPLICATION->metadataIndex()->get(comp->getID());
+                        versionList->waitToLoad();
+                        if (versionList) {
+                            auto recommended = versionList->getRecommendedForParent(uid, version);
+                            if (recommended) {
+                                qCDebug(instanceProfileC) << d->m_instance->name() << "|"
+                                                          << "setting updated loader to recommended version: " << comp->getID() << " = "
+                                                          << recommended->version();
+                                comp->setVersion(recommended->version());
+                            } else {
+                                auto latest = versionList->getLatestForParent(uid, version);
+                                if (latest) {
+                                    qCDebug(instanceProfileC) << d->m_instance->name() << "|"
+                                                              << "setting updated loader to latest compatible version: " << comp->getID()
+                                                              << " = " << latest->version();
+                                    comp->setVersion(latest->version());
+                                } else {
+                                    qCDebug(instanceProfileC) << d->m_instance->name() << "|"
+                                                              << "no compatible version for" << comp->getID() << "removing";
+                                    remove(comp->getID());
+                                }
+                            }
+                        } else {
+                            qCDebug(instanceProfileC) << d->m_instance->name() << "|"
+                                                      << "no version list in metadata index for" << comp->getID();
+                        }
+                    } else {
+                        qCDebug(instanceProfileC) << d->m_instance->name() << "|" << comp->getID() << ":" << comp->getVersion()
+                                                  << "linked to important component: " << uid << " | Removing so it re-resolves";
+                        remove(comp->getID());
```

Comment:
The removal of the component usually creates a visual glitch if you are on the Versions page where the component removed is still there but without a version.

Reply:
I'm not sure what the alternative is supposed to be, personally I haven't seen this visual glitch because it get re-resolved almost instantly

Reply:
Yeah I'm not experiencing this as often as before you can ignore it